// react-nscoder-footer/FooterBar.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function FooterBar(_ref) {
  var bgColor = _ref.bgColor,
      textcolor = _ref.textcolor,
      headlineColor = _ref.headlineColor,
      bordercolor = _ref.bordercolor,
      hoverTextColor = _ref.hoverTextColor,
      firstHeadlineName = _ref.firstHeadlineName,
      firstROWitems = _ref.firstROWitems,
      secondHeadlineName = _ref.secondHeadlineName,
      secondROWitems = _ref.secondROWitems,
      threeHeadlineName = _ref.threeHeadlineName,
      threedROWitems = _ref.threedROWitems,
      fourHeadlineName = _ref.fourHeadlineName,
      fourROWitems = _ref.fourROWitems,
      threeBottomItemsColor = _ref.threeBottomItemsColor,
      threeBottomItems = _ref.threeBottomItems;

  var _useState = (0, _react.useState)({}),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      ar = _useState2[0],
      setAr = _useState2[1];

  (0, _react.useEffect)(function () {
    return function () {
      setAr({});
    };
  }, []);

  function changeBgColor(a) {
    a.target.style.color = hoverTextColor;
  }

  function changeDefautColor(b) {
    b.target.style.color = textcolor;
    b.target.style.transition = '750ms';
  } //Three Bootom Items Default Color


  function changeDefautColorthreeBottomItems(c) {
    c.target.style.color = threeBottomItemsColor;
    c.target.style.transition = '750ms';
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    className: "mainFooter",
    style: {
      background: bgColor,
      color: textcolor
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "empty"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "firstDivFooter"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "firstDivFooterRow"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "firstDivFooterRowHeadline",
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor,
    style: {
      color: headlineColor,
      borderColor: bordercolor
    }
  }, firstHeadlineName), firstROWitems[0][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, firstROWitems[0][0]) : "", firstROWitems[1][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, firstROWitems[1][0]) : "", firstROWitems[2][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, firstROWitems[2][0]) : "", firstROWitems[3][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, firstROWitems[3][0]) : "", firstROWitems[4][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, firstROWitems[4][0]) : "", firstROWitems[5][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, firstROWitems[5][0]) : ""), /*#__PURE__*/_react.default.createElement("div", {
    className: "firstDivFooterRow"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "firstDivFooterRowHeadline",
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor,
    style: {
      color: headlineColor,
      borderColor: bordercolor
    }
  }, secondHeadlineName), secondROWitems[0][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, secondROWitems[0][0]) : "", secondROWitems[1][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, secondROWitems[1][0]) : "", secondROWitems[2][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, secondROWitems[2][0]) : "", secondROWitems[3][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, secondROWitems[3][0]) : "", secondROWitems[4][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, secondROWitems[4][0]) : "", secondROWitems[5][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, secondROWitems[5][0]) : ""), /*#__PURE__*/_react.default.createElement("div", {
    className: "firstDivFooterRow"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "firstDivFooterRowHeadline",
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor,
    style: {
      color: headlineColor,
      borderColor: bordercolor
    }
  }, threeHeadlineName), threedROWitems[0][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, threedROWitems[0][0]) : "", threedROWitems[1][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, threedROWitems[1][0]) : "", threedROWitems[2][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, threedROWitems[2][0]) : "", threedROWitems[3][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, threedROWitems[3][0]) : "", threedROWitems[4][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, threedROWitems[4][0]) : "", threedROWitems[5][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, threedROWitems[5][0]) : ""), /*#__PURE__*/_react.default.createElement("div", {
    className: "firstDivFooterRow"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "firstDivFooterRowHeadline",
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor,
    style: {
      color: headlineColor,
      borderColor: bordercolor
    }
  }, fourHeadlineName), fourROWitems[0][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, fourROWitems[0][0]) : "", fourROWitems[1][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, fourROWitems[1][0]) : "", fourROWitems[2][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, fourROWitems[2][0]) : "", fourROWitems[3][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, fourROWitems[3][0]) : "", fourROWitems[4][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, fourROWitems[4][0]) : "", fourROWitems[5][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColor
  }, fourROWitems[5][0]) : "")), /*#__PURE__*/_react.default.createElement("div", {
    className: "secondDiv",
    style: {
      color: threeBottomItemsColor
    }
  }, threeBottomItems[0][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColorthreeBottomItems
  }, threeBottomItems[0][0]) : "", threeBottomItems[1][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColorthreeBottomItems
  }, threeBottomItems[1][0]) : "", threeBottomItems[2][1] ? /*#__PURE__*/_react.default.createElement("div", {
    onMouseOver: changeBgColor,
    onMouseLeave: changeDefautColorthreeBottomItems
  }, threeBottomItems[2][0]) : ""), /*#__PURE__*/_react.default.createElement("div", {
    className: "empty"
  }));
}

var _default = FooterBar;
exports.default = _default;
